src/runs.rs

```diff
@@ -0,0 +1,341 @@
+            None => {
+                // We generate a random name for this run
+                let mut generator = Generator::default();
+                generator.next().expect("Failed to generate random name")
+            }
+        };
+
+        // Check not the same name as an existing run
+        if self.runs().iter().any(|run| run.name == name) {
+            miette::bail!("A run with the same name already exists. You can call `pixi runs clear` to clear all the terminated runs.");
+        }
+
+        Ok(DaemonRun::new(
+            name,
+            self.runs_dir(),
+            self.project.root().to_path_buf(),
+        ))
+    }
+
+    pub fn get_run(&self, name: String) -> miette::Result<DaemonRun> {
+        let run = DaemonRun::new(name, self.runs_dir(), self.project.root().to_path_buf());
+
+        // Check the pid file exists
+        if !run.pid_file_path().exists() {
+            miette::bail!("No run with name '{}' found.", run.name);
+        }
+
+        Ok(run)
+    }
+}
+
+#[derive(Debug)]
+pub struct DaemonRun {
+    pub name: String,
+    pub runs_dir: PathBuf,
+    pub working_dir: PathBuf,
+}
+
+impl DaemonRun {
+    pub fn new(name: String, runs_dir: PathBuf, working_dir: PathBuf) -> Self {
+        Self {
+            name,
+            runs_dir,
+            working_dir,
+        }
+    }
+
+    pub fn is_running(&self) -> bool {
+        !matches!(
+            self.process_status(),
+            DaemonRunStatus::Terminated | DaemonRunStatus::UnknownPid
+        )
+    }
+
+    pub fn process_status(&self) -> DaemonRunStatus {
+        let pid = match self.read_pid() {
+            Some(pid) => pid,
+            None => return DaemonRunStatus::UnknownPid,
+        };
+
+        // TODO: not very efficient to call this every time
+        let mut system = System::new_all();
+        system.refresh_all();
+
+        match system.process(pid) {
+            Some(process) => DaemonRunStatus::from_process_status(process.status()),
+            // if no process is associated with the pid, it means the process is terminated
+            None => DaemonRunStatus::Terminated,
+        }
+    }
+
+    pub fn pid_file_path(&self) -> PathBuf {
+        self.runs_dir.join(format!("{}.pid", self.name))
+    }
+
+    pub fn stdout_path(&self) -> PathBuf {
+        self.runs_dir.join(format!("{}.out", self.name))
+    }
+
+    pub fn stderr_path(&self) -> PathBuf {
+        self.runs_dir.join(format!("{}.err", self.name))
+    }
+
+    pub fn infos_file_path(&self) -> PathBuf {
+        self.runs_dir.join(format!("{}.infos.json", self.name))
+    }
+
+    pub fn read_pid(&self) -> Option<Pid> {
+        if !self.pid_file_path().exists() {
+            return None;
+        }
+
+        let pid = std::fs::read_to_string(self.pid_file_path())
+            .expect("Failed to read pid file")
+            .trim()
+            .parse::<Pid>()
+            .expect("Failed to parse pid file content as u32");
```

Comment:
If a user manually edited this fail, pixi could crash. Instead I think we should just return `None` or an error.

Reply:
yes good point - will return an error